Application/Items/Commands/Redeem.cs

```diff
@@ -39,12 +39,12 @@ public async Task<Unit> Handle(Command request, CancellationToken cancellationTo
                     throw new ConflictException($"Item {request.Id} has already been redeemed");
                 }
 
-                if (item.Uploader == request.User)
+                if (item.Uploader.Username == request.User)
                 {
                     throw new ConflictException($"Item {request.Id} belongs to the client, therefore can not be redeemed");
                 }
 
-                item.Redeemer = request.User;
+                item.Redeemer = user;
```

Comment:
check the value of request.item.uploader, because I think the wrong data is assigned